### ISSUE TITLE
Current builds are broken for some users due to outdated go binaries

### ISSUE THREAD

**Issue by sharms** [cloudfoundry/cli#1080]
## Command  

`cf login`

## What occurred

```
cf login -a api.mycf.com --sso
API endpoint: api.mycf.com
FAILED
Error performing request: Get https://api.mycf.com/v2/info: dial tcp: i/o timeout
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.
```

## What you expected to occur
This is the output when I recompile using Go 1.8:

```
go run main.go login -a api.mycf.com --sso
API endpoint: api.mycf.com

One Time Code ( Get one at https://login.mycf.com/passcode )>
```
## CLI Version  

cf version 6.24.0+c7a32d3.2017-02-10

## Platform & Shell Details  

Ubuntu 14.04

## Any other relevant information  
CF CLI uses native go DNS resolution, which has many outstanding bugs in older versions.  Go 1.8 solves this errant case where it attempts to connect to ipv6 when no ipv6 is available.

**Comment by sharms** [comment]
Specifically maybe this shouldn't be hard coded to 1.5.1: https://hub.docker.com/r/cloudfoundry/cli-ci/tags/

**Comment by mmb** [comment]
The 1.5.1 is an old tag. The go version in the container is currently 1.7.4.

```
docker run cloudfoundry/cli-ci go version
go version go1.7.4 linux/amd64
```

**Comment by sharms** [comment]
Ok - at any rate, go1.8 fixes DNS resolution in which go receives a AAAA record first, attempts to resolve it on a ipv4 system, and times out.

Either this or enable native DNS resolution, but upgrading to 1.8 is more straight forward and has other tangible bug benefits.

**Comment by dkoper** [comment]
@sharms Do you have a link to the Golang issue that is causing your issue with the cf CLI that is fixed in 1.8?
Also, does tuning CF_DIAL_TIMEOUT not have an impact?

We have previously always upgraded golang to the latest right after its release, and every time there were unpleasant surprises. (Several times related to DNS).
This time, although acknowledging there are benefits, we're not going to rush.
We have a story in the backlog to upgrade to golang 8.1 when it comes out. If we find a bug fix or feature in 1.8.0 that we really want, we can consider upgrading sooner.

**Comment by dkoper** [comment]
Closing. We found one bug report for Golang 1.8.0 that may affect the CLI. Planning to wait for 1.8.1 unless we find a feature in 1.8.0 that we can't wait for, in which case we can give the bug report a closer look.
